firebase-firestore/ktx/src/test/kotlin/com/google/firebase/firestore/ktx/NestedMapEncoderTests.kt

```diff
@@ -0,0 +1,308 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx
+
+import com.google.firebase.firestore.ktx.serialization.encodeToMap
+import kotlin.test.assertFailsWith
+import kotlinx.serialization.Serializable
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class NestedMapEncoderTests {
+
+    @Test
+    fun `plan custom object encoding is supported`() {
+        @Serializable data class PlainProject(val name: String, val ownerName: String)
+        val plainObject = PlainProject("kotlinx.serialization", "kotlin")
+        val encodedMap = encodeToMap(plainObject)
+        val expectedMap = mapOf("name" to "kotlinx.serialization", "ownerName" to "kotlin")
+        assertTrue(encodedMap == expectedMap)
```

Comment:
You can use https://truth.dev/ to write assertions too. It generates more readable errors and code

Reply:
I changed the `assertTrue` to asserThat` now for all my unit test and integration test now.

Reply:
Are you planning on updating it here?